modules/google-maps/src/google-maps-overlay.ts

```diff
@@ -179,11 +204,8 @@ export default class GoogleMapsOverlay {
 
   // Vector code path
   _onDrawVectorInterleaved({gl, transformer}) {
-    if (!this._deck || !this._map) {
-      return;
-    }
-
     const deck = this._deck;
+    assert(this._map && this._overlay && deck);
```

Comment:
This is a breaking change. Where it silently skipped before you are now crashing the app. Is it necessary?

Reply:
Btw, I've discussed it in [already collapsed comment](https://github.com/visgl/deck.gl/pull/6835#discussion_r853017297).

I see that in #6803 You've chose `// @ts-ignore (TS2531) state is always defined` method -  i will adapt to this way of working-around Typescript strictness.

Rest of comment only for reference.

> This is a breaking change (..) Where it silently skipped before ...

I know that excessive usage of `assert` have consequences and i'll try to reduce it, but it's not _hard_ breaking change i.e would  `this._map && this._overlay && deck` be falsy, this method would crash by with old "cannot access 'x' of undefined/null" at first random access to `deck` or to `this._map`.
So, it only changes signature of crash.

(Yes, `this._overlay` check was excessive copy-pasta, sorry)

Reply:
I'm not sure I follow. Your previous comment was about the use of assert in the `getViewPropsFromOverlay` utility. This particular conversation is regarding you deleting

```js
if (!this._deck || !this._map) {
      return;
    }
```

And adding

```js
assert(this._map && this._overlay && deck);
```

How is it not a breaking change?

Referencing original PR #6340

Reply:
@Pessimistress 
I am sorry for deleting comment, it was mistake from my side and i deleted to prevent "misinformation and wasted effort" it as soon as i've realized it's bogus.

---

You're absolutely right, this particular assertion was changing behavior. I've managed to remove most all of them by changing approach to "no-op" in callbacks.

Here's my original comment for reference:

---
Btw, I've discussed it in already collapsed comment.

I see that in #6803 You've chose // @ts-ignore (TS2531) state is always defined method - i will adapt to this way of working-around Typescript strictness.

Rest of comment only for reference.

This is a breaking change (..) Where it silently skipped before ...

I know that excessive usage of assert have consequences and i'll try to reduce it, but it's not hard breaking change i.e would this._map && this._overlay && deck be falsy, this method would crash by with old "cannot access 'x' of undefined/null" at first random access to deck or to this._map.
So, it only changes signature of crash.

(Yes, this._overlay check was excessive copy-pasta, sorry)